main.py

```diff
@@ -158,3 +170,31 @@ def fetch_and_store():
 
     return {"message": "Data successfuly stored in the database"}
 
+
+@app.get("/pokemon/")
+def get_pokemon(sort: str = Query("asc", description="Sort order: 'asc' or 'desc'"),
+                keyword: str = Query(None, description="Search keyword"),
+                col: str = Query("name", description="Column to search in, default 'name'"),
+                limit: int = Query(10, description="Results per page, default 10", le=100),
+                page: int = Query(1, description="Page number")):
+    # Verify column exists
+    if not hasattr(models.PokemonData, col):
```

Comment:
Along with column check, you need to check if the keyword datatype and column datatype match
you can't search "abc" in the below columns

total: int
hp: int
attack: int
defense: int
sp_atk: int
sp_def: int
speed: int
generation: int
legendary: bool

similarly, you might need to convert int inputs to string when searching in string-type columns
name: str
type_1: str
type_2: str